Modules/_functoolsmodule.c

```diff
@@ -748,25 +763,37 @@ typedef struct lru_list_elem {
     PyObject *key, *result;
 } lru_list_elem;
 
+static int
+lru_list_elem_traverse(lru_list_elem *link, visitproc visit, void *arg)
+{
+    Py_VISIT(link->key);
+    Py_VISIT(link->result);
+    Py_VISIT(Py_TYPE(link));
+    return 0;
+}
```

Comment:
A lru_list_elem_clear() function would also be needed to implement the GC protocol, no?

Reply:
> A lru_list_elem_clear() function would also be needed to implement the GC protocol, no?

I was under the impression traverse was enough. I may be wrong.

Reply:
Without clear, the ref count cannot each zero, and the GC cannot break cycles: https://devguide.python.org/garbage_collector/#destroying-unreachable-objects

cc @pablogsal

Reply:
`_lru_list_elem` is not a standalone type. It is an internal of `_lru_cache_wrapper`.

* `lru_cache_tp_traverse()` visits `lru_list_elem`.
* `lru_cache_tp_clear()` clears `lru_list_elem`.

That's why we can keep `_lru_list_elem` non-GC type.

Reply:
IMO, Inada's #26416 is acceptable.